src/main/java/org/dspace/pack/bagit/BagItAipReader.java

```diff
@@ -0,0 +1,287 @@
+            lines = Collections.emptyList();
+        }
+
+        return lines;
+    }
+
+    /**
+     * Retrieve the logo for an aip, if it is available
+     *
+     * @return an {@link Optional} containing the {@link InputStream} of the logo if it exists, and
+     * {@link Optional#absent()} otherwise
+     * @throws IOException if there is an error getting the {@link InputStream} for the logo
+     */
+    public Optional<Path> findLogo() throws IOException {
+        // Search for the bitstream in the data directory
+        final DirectoryStream.Filter<Path> bitstreamFilter = new DirectoryStream.Filter<Path>() {
+            @Override
+            public boolean accept(Path path) {
+                // checking start with on the Path doesn't really work as expected, so get the relative name and make
+                // it a String
+                final String filename = path.getFileName().toString();
+                return path.toFile().isFile() && filename.startsWith("bitstream");
+            }
+        };
+
+        Optional<Path> logo = Optional.absent();
+        try(DirectoryStream<Path> bitstreams = Files.newDirectoryStream(bag.resolve(dataDirectory), bitstreamFilter)) {
+            final Iterator<Path> iterator = bitstreams.iterator();
+            if (iterator.hasNext()) {
+                logo = Optional.of(iterator.next());
+            }
+        }
+
+        return logo;
+    }
+
+    /**
+     * Read the metadata.xml file located in a bags data directory
+     *
+     * @return a list of {@link XmlElement}s which were read from the file
+     * @throws IOException if there was an error reading the file or parsing the xml
+     */
+    public List<XmlElement> readMetadata() throws IOException {
+        final Path metadata = bag.resolve(metadataLocation);
+        try (InputStream metadataStream = Files.newInputStream(metadata)) {
+            return readXml(metadataStream);
+        }
+    }
+
+    /**
+     * Find any {@link org.dspace.content.Bitstream}s packaged in an aip and return path, bundle name, and metadata
+     *
+     * @return a list of {@link PackagedBitstream}s found
+     * @throws IOException if there are any errors searching for bitstreams
+     */
+    public List<PackagedBitstream> findBitstreams() throws IOException {
+        final Path data = bag.resolve(dataDirectory);
+
+        // filter to find only directories (for bundle names)
+        final DirectoryStream.Filter<Path> directoryFilter = new DirectoryStream.Filter<Path>() {
+            @Override
+            public boolean accept(Path path) {
+                return Files.isDirectory(path);
+            }
+        };
+
+        // filter to find only the bitstream file
+        final DirectoryStream.Filter<Path> bitstreamFilter = new DirectoryStream.Filter<Path>() {
+            @Override
+            public boolean accept(Path path) {
+                return path.toFile().isFile() && !path.getFileName().toString().endsWith(xmlSuffix);
```

Comment:
Pretty sure this will be a bug if the bitstream is also an xml file. During the initial dev the bitstreams didn't have a file extension, but now that this likely needs to be updated.